test_api.py

```diff
@@ -155,8 +155,8 @@ class MockEmitEvent:
     def __init__(self):
         self.events = []
 
-    def __call__(self, e):
-        self.events.append(e)
+    def __call__(self, e, id):
+        self.events.append((e, id))
```

Comment:
Shouldn't id be part of the event instead of in a tuple with the event? Seems like it'd avoid all of the extra subscripting and be more true to how events are actually returned unless I'm missing something here

Reply:
Can you elaborate a little bit here? Key isn't part of the event right now ([here](https://github.com/RedHatInsights/insights-host-inventory/pull/583/files#diff-53f0a5ac005ca800cc1e581e9e303bd4R33)). Making `write_event` take only `event` instead of `event` and `host_id` will require parsing it back from a JSON obj to a dictionary. I am sure if that would be optimal if I did it... :thinking: 
`event` gets converted to a JSON in `build_event()` of `egress.py`

Reply:
I see. Disregard then, I thought the event included the key

Reply:
Cool! :+1: :+1:

Reply:
The event is only the actual payload. It may make sense though to make this an Event struct (namedtuple) with _payload_ and _key_ properties. A plain tuple should be fine here though.

```suggestion
    def __call__(self, e, id):
        self.events.append((e, key))
```